transform-plugins/src/main/java/io/cdap/plugin/UnionSplitter.java

```diff
@@ -174,42 +177,45 @@ public void transform(StructuredRecord record, MultiOutputEmitter<StructuredReco
 
     Schema.Field unionSchemaField = inputSchema.getField(unionField);
     if (unionSchemaField == null) {
-      throw new IllegalArgumentException(
-        String.format("Field '%s' does not exist in the input schema.", unionField));
-    }
-    Schema unionSchema = unionSchemaField.getSchema();
-    if (unionSchema.getType() != Schema.Type.UNION) {
-      throw new IllegalArgumentException(
-        String.format("Field '%s' is not of type union, but is of type '%s'", unionField, unionSchema.getType()));
-    }
-
-    int numFields = inputSchema.getFields().size();
-    ArrayList<Schema.Field> outputFields = new ArrayList<>(numFields);
-    int i = 0;
-    int unionIndex = -1;
-    for (Schema.Field inputField : inputSchema.getFields()) {
-      if (inputField.getName().equals(unionField)) {
-        unionIndex = i;
-        outputFields.add(null);
+      collector.addFailure(String.format("Field '%s' must exist in input schema.", unionField), null)
+        .withConfigProperty(Conf.UNION_FIELD);
+    } else {
+      Schema unionSchema = unionSchemaField.getSchema();
+      if (unionSchema.getType() != Schema.Type.UNION) {
+        collector.addFailure(String.format("Field '%s' must be of type union, but is of type '%s'.",
+                                           unionField, unionSchema.getType()), null)
+          .withConfigProperty(Conf.UNION_FIELD);
       } else {
-        outputFields.add(inputField);
+        int numFields = inputSchema.getFields().size();
+        ArrayList<Schema.Field> outputFields = new ArrayList<>(numFields);
+        int i = 0;
+        int unionIndex = -1;
+        for (Schema.Field inputField : inputSchema.getFields()) {
+          if (inputField.getName().equals(unionField)) {
+            unionIndex = i;
+            outputFields.add(null);
+          } else {
+            outputFields.add(inputField);
+          }
+          i++;
+        }
+
+        for (Schema schema : unionSchema.getUnionSchemas()) {
+          Schema.Type type = schema.getType();
+          switch (type) {
+            case ENUM:
+            case MAP:
+            case ARRAY:
+            case UNION:
+              collector.addFailure(String.format("Unsupported type '%s' within a union.", type), null)
```

Comment:
I'm not sure these should be separate failures for each type. It seems more fitting to use a single failure, with the message saying which types are invalid.

Reply:
Added a corrective action to indicate what the unsupported types are. We can also remove the specific typing message if that's desired.

Reply:
what I meant here is that since this is in a for loop, a field with multiple invalid types in its union will result in multiple failures. This should instead be a single failure, with the message mentioning which types are invalid.

Reply:
Gotcha, fixed this by using if statement coupled with a break statement upon discovering a failure.